packages/flutter_tools/lib/src/asset.dart

```diff
@@ -452,26 +469,40 @@ class ManifestAssetBundle implements AssetBundle {
         _fileSystem.file('DOES_NOT_EXIST_RERUN_FOR_WILDCARD$suffix').absolute);
     }
 
-    _setIfChanged(_kAssetManifestJson, assetManifest, AssetKind.regular);
+    _setIfChanged(_kAssetManifestJsonFilename, assetManifestJson, AssetKind.regular);
+    _setIfChanged(_kAssetManifestBinFilename, assetManifestBinary, AssetKind.regular);
     _setIfChanged(kFontManifestJson, fontManifest, AssetKind.regular);
     _setLicenseIfChanged(licenseResult.combinedLicenses, targetPlatform);
     return 0;
   }
 
   @override
   List<File> additionalDependencies = <File>[];
+  void _setIfChanged(String key, DevFSContent content, AssetKind assetKind) {
+    bool areEqual(List<int> o1, List<int> o2) {
+      if (o1.length != o2.length) {
+        return false;
+      }
 
-  void _setIfChanged(String key, DevFSStringContent content, AssetKind assetKind) {
-    if (!entries.containsKey(key)) {
-      entries[key] = content;
-      entryKinds[key] = assetKind;
-      return;
+      for (int index = 0; index < o1.length; index++) {
+        if (o1[index] != o2[index]) {
+          return false;
+        }
+      }
+
+      return true;
     }
```

Comment:
I thought we might already have a helper for this somewhere in flutter_tools, but if not I'd pull this into a private static method.

Reply:
I did a quick regex search for such a function and didn't find any. In a previous version of this, I brought in the `ListCompare` utility from the `collections` package, but I chose not to do that this time around to avoid adding `collections` as a dependency in `flutter_tools`.

Out of curiosity, why do you prefer a private static method here? I made `areEqual` an inner function to keep it's scope as narrow as possible (I only wish inner functions could be declared after use in Dart).

Reply:
the function is available in a narrower scope, but now it closes over all of the locals in that method so its own scope is bigger. Prefer private methods, library scope is limited enough

Reply:
> but now it closes over all of the locals in that method so its own scope is bigger

Ah, I failed to consider that--good call.